### PR TITLE
fix: Ensure backward compatibility w.r.t. statusCode and response.body in case of errors

### LINKED ISSUE #20438: [Task]: Modify appsmith error messages for all the plugins
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description
All the current error messages for each plugin are listed in this [Google Sheet](https://docs.google.com/spreadsheets/d/1EDunPez23Tz1YBvxtf4GsQC7y-DrqCSK5qghGmPLE1I/edit#gid=0). 

### SubTasks

_No response_

### PR DESCRIPTION
**Pull request by sondermanish**
## Description
Since we have introduced the error handling framework, each plugin produces error messages specific to the plugin execution exception. This Pr is just about standardising these error messages. 
In this PR we are handling only five plugins as mentioned:
- Rest API
- MongoDB
- Postgres
- MySql
- Google Sheets

> Modifying error messages for five plugins Rest API, Mongo, Postgres, MySQL, and Google Sheets for improving consistency. 

In addition it also takes care of the following issues
- If the downstream error message is available response.body will hold that value and it will hold the appsmith error message otherwise
- Don't override the actual http status code with Appsmith error code in case of RestAPI and GraphQL plugins

Fixes #20438 #21022

## Type of change
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/constants/ErrorMessages.java [hunk 1]
```diff
@@ -2,35 +2,35 @@
 
 public class ErrorMessages {
 
-    public static final String EMPTY_ROW_OBJECT_MESSAGE = "Row Object cannot be empty. Please edit the Row Object field in the Google Sheet query form.";
+    public static final String EMPTY_ROW_OBJECT_MESSAGE = "Row object(s) cannot be empty.";
 
-    public static final String EMPTY_ROW_ARRAY_OBJECT_MESSAGE = "Row Object(s) cannot be empty. Please edit the Row Object(s) field in the Google Sheet query form.";
+    public static final String EMPTY_ROW_ARRAY_OBJECT_MESSAGE = "Row object(s) cannot be empty.";
```
Comment (subrata71) [inline app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/constants/ErrorMessages.java:7]: These two are essentially the same.

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/constants/ErrorMessages.java [hunk 2]
```diff
@@ -2,35 +2,35 @@
...
 
-    public static final String EMPTY_UPDATE_ROW_OBJECT_MESSAGE = "Update Row Object cannot be empty. Please edit the Update Row Object field in the Google Sheet query form.";
+    public static final String EMPTY_UPDATE_ROW_OBJECT_MESSAGE = "Update Row object(s) cannot be empty.";
 
-    public static final String EMPTY_UPDATE_ROW_OBJECTS_MESSAGE = "Update Row Object(s) cannot be empty. Please edit the Update Row Object(s) field in the Google Sheet query form.";
+    public static final String EMPTY_UPDATE_ROW_OBJECTS_MESSAGE = "Update Row object(s) cannot be empty.";
```
Comment (subrata71) [inline app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/constants/ErrorMessages.java:11]: These two are essentially the same.

**Review by subrata71 — CHANGES_REQUESTED** [review]

File: app/server/appsmith-plugins/postgresPlugin/src/main/java/com/external/plugins/exceptions/PostgresErrorMessages.java [hunk 3]
```diff
@@ -6,22 +6,20 @@ private PostgresErrorMessages() {
     }
     public static final String MISSING_QUERY_ERROR_MSG = "Missing required parameter: Query.";
 
-    public static final String QUERY_EXECUTION_FAILED_ERROR_MSG = "Your PostgreSQL query failed to execute. Please check more information in the error details.";
+    public static final String QUERY_EXECUTION_FAILED_ERROR_MSG = "Your PostgreSQL query failed to execute.";
 
-    public static final String POSTGRES_JDBC_DRIVER_LOADING_ERROR_MSG = "Error loading Postgres JDBC Driver class.";
+    public static final String POSTGRES_JDBC_DRIVER_LOADING_ERROR_MSG = "Your PostgreSQL query failed to execute.";
```
[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-plugins/postgresPlugin/src/main/java/com/external/plugins/exceptions/PostgresErrorMessages.java:11]: Although this doesn't make any diff on the end result and messages, the variable name itself improves code readability.

[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-plugins/postgresPlugin/src/main/java/com/external/plugins/exceptions/PostgresErrorMessages.java:11]: Resolving conversation as discussed.

**Review by subrata71 — APPROVED** [review]

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress changes looks ok!

**Pull request merged**